### PR TITLE
Implement Search Requests API

### LINKED ISSUE #439: Implement Requests API for Searcher
### Summary

Search API contains methods to access the Suggestions database.

### Value

Search API allows users to query the Suggestions database for the autocomplete items.

### Specification

- [ ] implement `search/getSuggestionsDatabase` method.
- [ ] implement `search/getSuggestionsDatabaseVersion` method.
- [ ] implement `search/completion` method.

### Acceptance Criteria & Test Cases

- All requests for the Searcher API are implemented.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #439

PR implements the search requests API and wires them to the database.

Changelog;

- add: `search/getSuggestionsDatabase` request
- add: `search/getSuggestionsDatabaseVersion` request
- add: `search/completion` request
- add: impl language-server part of the integraton
  
### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

**Review by kustosz — APPROVED** [review]
Cool stuff! Approval is conditional on fixing the initialization flow.

File: lib/searcher/src/main/scala/org/enso/searcher/sql/SqlSuggestionsRepo.scala [hunk 1]
```diff
@@ -16,38 +17,139 @@ final class SqlSuggestionsRepo(implicit ec: ExecutionContext)
...
+    joined
+      .filterOpt(selfType) {
+        case (row, value) => row._2.selfType === value
+      }
+      .filterOpt(returnType) {
+        case (row, value) => row._2.returnType === value
+      }
+      .filterOpt(kinds) {
+        case (row, value) => row._2.kind inSet value.map(SuggestionKind(_))
+      }
```
Comment (lolczak) [inline lib/searcher/src/main/scala/org/enso/searcher/sql/SqlSuggestionsRepo.scala]: It is a potential bottleneck. It joins two tables and next performs a full table scan.

**Review by lolczak — COMMENTED** [review]
My big concern is that the usage of the SQL database is a potential bottleneck. The main query that searches suggestions joins two tables and performs a full table scan. For a big project, it can be very slow and can degrade the user experience. I suggest rethinking the underlying storage technology and consider using an inverted index like Apache Lucene that is more performant in terms of querying. Perhaps performance tests are needed.

File: lib/searcher/src/main/scala/org/enso/searcher/sql/SqlSuggestionsRepo.scala (hunk 1 above)
Comment (4e6) [reply lib/searcher/src/main/scala/org/enso/searcher/sql/SqlSuggestionsRepo.scala]: Good point. I spent some time today, checking the query plans, adding logging and missing indexes. All of the search queries make use of indexes now.

Also, search query returns only `suggestions.id` field, and we don't need the joined query at all here. End up removing the join.

Below are the resulting SQL queries that are being executed:

```sql
explain query plan select "id" from "suggestions" where "return_type" = 'MyType';
QUERY PLAN
`--SEARCH TABLE suggestions USING COVERING INDEX suggestions_return_type_idx (return_type=?)
```

```sql
explain query plan select "id" from "suggestions" where "self_type" = 'Main';
QUERY PLAN
`--SEARCH TABLE suggestions USING COVERING INDEX suggestions_self_type_idx (self_type=?)
```

```sql
explain query plan select "id" from "suggestions" where "kind" in (0, 3);
QUERY PLAN
`--SEARCH TABLE suggestions USING COVERING INDEX suggestions_kind_idx (kind=?)
```

**Review by lolczak — CHANGES_REQUESTED** [review]
The `SqlSuggestionRepo` is a leaky abstraction. It should hide all details regarding the implementation and the underlying technology. Unfortunately, Slick objects leak out from the repo. The repository should be easily replaceable without modifications in clients that use the repo (but the repo needs accompanying DB object to run queries). In other words, the `SqlSuggestionRepo` should extend `SuggestionsRepo[Future]` or `SuggestionsRepo[IO]` not `SuggestionsRepo[DBIO]`. We may change the storage in the future. We should do it by implementing new repo.

**Comment by 4e6** [comment]
Benchmark results for a database of 1_000_000 records

```
[info] Benchmark                                         Mode  Cnt    Score    Error  Units
[info] SuggestionsRepoBenchmark.searchBaseline           avgt    5   94.562 ±  9.673  us/op
[info] SuggestionsRepoBenchmark.searchByAll              avgt    5  274.080 ± 12.816  us/op
[info] SuggestionsRepoBenchmark.searchByReturnType       avgt    5  220.701 ± 11.551  us/op
[info] SuggestionsRepoBenchmark.searchBySelfReturnTypes  avgt    5  242.471 ±  9.945  us/op
[info] SuggestionsRepoBenchmark.searchBySelfType         avgt    5  226.999 ± 17.085  us/op
```

**Comment by iamrecursion** [comment]
> Benchmark results for a database of 1_000_000 records
> 
> ```
> [info] Benchmark                                         Mode  Cnt    Score    Error  Units
> [info] SuggestionsRepoBenchmark.searchBaseline           avgt    5   94.562 ±  9.673  us/op
> [info] SuggestionsRepoBenchmark.searchByAll              avgt    5  274.080 ± 12.816  us/op
> [info] SuggestionsRepoBenchmark.searchByReturnType       avgt    5  220.701 ± 11.551  us/op
> [info] SuggestionsRepoBenchmark.searchBySelfReturnTypes  avgt    5  242.471 ±  9.945  us/op
> [info] SuggestionsRepoBenchmark.searchBySelfType         avgt    5  226.999 ± 17.085  us/op
> ```

I think this kind of performance is probably fine.

**Review by lolczak — CHANGES_REQUESTED** [review]
I don't like that the data access logic is scattered on two objects, both having the same scaladoc `/** The object for accessing the suggestions database. */`. It means that the design has low cohesion. The whole data access logic should be implemented in the repository object. Queries and joins are the essential part of repositories and should sit in the repo class.

**Review by lolczak — APPROVED** [review]

**Pull request merged**